lattices/src/set_union_with_tombstones.rs

```diff
@@ -0,0 +1,330 @@
+//! Module containing the [`SetUnionWithTombstones`] lattice and aliases for different datastructures.
+
+use std::cmp::Ordering::{self, *};
+use std::collections::{BTreeSet, HashSet};
+
+use cc_traits::{Collection, Get, Remove};
+
+use crate::cc_traits::{Iter, Len, Set};
+use crate::collections::{ArraySet, EmptySet, OptionSet, SingletonSet};
+use crate::{IsBot, IsTop, LatticeFrom, LatticeOrd, Merge};
+
+/// Set-union lattice with tombstones.
+///
+/// When an item is deleted from the SetUnionWithTombstones, it is removed from `set` and added to `tombstones`.
+/// This also is an invariant, if an item appears in `tombstones` it must not also be in `set`.
+///
+/// Merging set-union lattices is done by unioning the keys of both the (set and tombstone) sets,
+/// and then removing any items in `set` that now exist in `tombstones`
```

Comment:
This sounds like set XOR when it should be set difference?

Reply:
hmm, that is not what I'm trying to convey. it should be set subtraction, set = set - tombstones, that is the invariant and operation that is performed.